### ISSUE TITLE
Second NewClient() overrides initial NewClient()?

### ISSUE THREAD

**Issue by darrenparkinson** [jbogarin/go-cisco-webex-teams#20]
Hi, me again ;)

I'm experiencing a weird issue which I suspect is obvious, but I can't quite get my head around it...

Essentially, if I create two clients with `NewClient()` using two separate tokens with `SetAuthToken()` the second one seems to overwrite the first.

For example, this code works as expected:

```go
	client1 := webexteams.NewClient()
	client1.SetAuthToken("first token here")
	me1, _, _ := client1.People.GetMe()
	fmt.Println(me1.DisplayName)
	
	client2 := webexteams.NewClient()
	client2.SetAuthToken("second token here")
	me2, _, _ := client2.People.GetMe()
	fmt.Println(me2.DisplayName)
```

But this does not, it basically prints the second clients display name twice

```go
	client1 := webexteams.NewClient()
	client1.SetAuthToken("first token here")
	client2 := webexteams.NewClient()
	client2.SetAuthToken("second token here")

	me1, _, _ := client1.People.GetMe()
	fmt.Println(me1.DisplayName)
	me2, _, _ := client2.People.GetMe()
	fmt.Println(me2.DisplayName)
```

Any thoughts before I start digging into it further greatly appreciated. 

Thanks.

**Comment by darrenparkinson** [comment]
I think it might be the use of the global variable RestyClient which is overridden each time a new client is initialised and then used for API calls:

`var RestyClient *resty.Client`

**Comment by darrenparkinson** [comment]
I tried the following changes which seemed to work.  Any chance you could try it out and maybe implement?  Thanks.

`api-client.go`:

* modify `service` client to be `*resty.Client` instead of `*Client` (not sure if this gets used anywhere else?)
* update NewClient to add the result of resty.New() to `c.common.client` where `c := &Client{}`

```go
type service struct {
	client *resty.Client
}
```

```go
func NewClient() *Client {
	client := resty.New()
        client.SetHostURL(apiURL)
	if os.Getenv("WEBEX_TEAMS_ACCESS_TOKEN") != "" {
		client.SetAuthToken(os.Getenv("WEBEX_TEAMS_ACCESS_TOKEN"))
	}

	c := &Client{}
	c.common.client = client

        ...
}
```

then in each of the services, e.g. `people_api.go`:

* replace instances of `RestyClient.R()` with `s.client.R()`

```go
func (s *PeopleService) GetMe() (*Person, *resty.Response, error) {

	path := "/people/me"

	response, err := s.client.R().
		SetResult(&Person{}).
		SetError(&Error{}).
		Get(path)

	if err != nil {
		return nil, nil, err
	}

	result := response.Result().(*Person)
	return result, response, err

}
```

I also had to update the pagination functions to make them methods of the service and then update the List methods to reference it under s.

**Comment by jbogarin** [comment]
@darrenparkinson Sorry for the delay. Let me test this tomorrow.

**Comment by jbogarin** [comment]
@darrenparkinson Thanks a lot for your valuable input. I tested your recommendations and included them in the repo, plus I created tag 0.4.2
